Lua/plugin.lua

```diff
@@ -15,8 +15,8 @@ function OnSetText(uri, text)
 	end
 
 	-- prevent diagnostic errors from in unpacking (a, b, c in t)
-	for vars, inPos, afterInPos, tablePos, tableName, finishPos in text:gmatch '([_%w, ]*)%s+()in()%s+()([_%w]*)()' do
-		if tableName ~= 'ipairs' and tableName ~= 'pairs' and vars:sub(1, 3) ~= 'for' then
+	for vars, inPos, afterInPos, tablePos, tableName, finishPos in text:gmatch '([_%w, ]*)%s+()in()%s+()([_%w]*%s-%(?.-%)?)()' do
+		if tableName ~= 'ipairs' and tableName ~= 'pairs' and not vars:find('^%s*for%s') then
```

Comment:
I couldn't actually think of any cases where `in` is used outside of pairs and ipairs, which were both being picked up when using `for k in pairs(tbl)` (where the second return value is undefined).

Checking for one or the other is probably preferred but my Lua patterns are unrefined.

Reply:
I feel like only the check for `for ...` should be kept, since it's a simple check and `for` is a reserved keyword. Users could name their variable `pairs`, for whatever reason, and then in unpacking won't work correctly for their custom `pairs` variable.

Reply:
Assuming you've resolved the issue when using pairs/ipairs, then remove the check and just use the pattern.
I didn't take a look but I assume this was the diffed output.
```diff
--for k in pairs(foo) do
++for k = pairs(foo) do
	print(k)
end
```

So just vars:find _should_ be enough, and revert the change to `@meta`.

Reply:
Now that I'm thinking about your actual issue was probably fixed by 842ecc2? Since when using `for` the statement would be ignored completely.